file_paths.md

```diff
@@ -0,0 +1,123 @@
+        client.cfg
+        facts.yaml
+        server.cfg
+
+    /etc/puppetlabs/puppet                # :confdir                     /etc/puppet
+        auth.conf                         # :rest_authconfig
+        autosign.conf                     # :autosign
+        binder_config.yaml                # :binder_config
+        csr_attributes.yaml               # :csr_attributes
+        custom_trusted_oid_mapping.yaml   # :trusted_oid_mapping_file
+        device.conf                       # :deviceconfig
+        fileserver.conf                   # :fileserverconfig
+        puppet.conf                       # :config
+        routes.yaml                       # :route_file
+        ssl                               # :ssldir                      /etc/puppet/ssl
+
+    /opt/puppetlabs/bin                   # symlink targets of puppet related binaries, excluding ruby, openssl
+        cfacter@                          -> /opt/puppetlabs/puppet/bin/cfacter
+        facter@                           -> /opt/puppetlabs/puppet/bin/facter
+        hiera@                            -> /opt/puppetlabs/puppet/bin/hiera
+        mco@                              -> /opt/puppetlabs/puppet/bin/mco
+        mcollectived@                     -> /opt/puppetlabs/puppet/bin/mcollectived
+        puppet@                           -> /opt/puppetlabs/puppet/bin/puppet
+
+    /opt/puppetlabs/mcollective/
+        plugins
+
+    /opt/puppetlabs/puppet                # ruby-puppet root
+        bin
+            cfacter
+            facter
+            gem
+            hiera
+            mco
+            mcollectived
+            openssl
+            puppet
+            ruby
+        cache                             # :vardir                      /var/lib/puppet
+            bucket                        # :bucketdir
+            client_yaml                   # :clientyamldir
+            client_data                   # :client_datadir
+            clientbucket                  # :clientbucketdir
+            devicedir                     # :devices
+            facts.d                       # :pluginfactdest
+            lib                           # :libdir
+            facts                         # used to generate :factpath
+            puppet-module                 # :module_working_dir
+            reports                       # :reportdir
+            server_datadir                # :server_data
+            state                         # :statedir
+            yaml                          # :yamldir
+        include
+            facter
+            openssl
+        lib
+            libaugeas.so
+            libcrypto.so
+            libfacter.so
+            libruby.so
+            libssl.so
+            ruby
+                vendor_ruby               # ruby code
+                    cfacter.rb
+                    facter.rb
+                    hiera.rb
+                    mcollective.rb
+                    puppet.rb
+            virt-what
```

Comment:
That matches distro packages. They have that file in lib, and a binary in sbin or bin.